File: 0792-binary-search/solution.cpp
class Solution {
public:
    int binarySearch(vector<int>& arr, int start, int end, int target) {
        if (start > end) 
            return -1;  // base case: not found

        int mid = start + (end - start) / 2;

        if (arr[mid] == target)
            return mid;

        if (arr[mid] < target)
            return binarySearch(arr, mid + 1, end, target);  // search right side
        else
            return binarySearch(arr, start, mid - 1, target); // search left side
    }

    int search(vector<int>& arr, int target) {
        return binarySearch(arr, 0, arr.size() - 1, target);
    }
};

File: 0792-binary-search/solution.java
class Solution {

    static int helper(int arr[],int s,int e,int target,int ans){
        if(s>e) return ans;

        int mid = s + (e-s)/2;

        if(arr[mid] == target){
            ans = mid;
            return ans;
        }

        else if(arr[mid] < target) return helper(arr,mid+1,e,target,ans);
        else return helper(arr,s,mid-1,target,ans);
        
    }

    public int search(int[] nums, int target) {
        return helper(nums,0,nums.length-1,target,-1);
    }
}
